### ISSUE TITLE
Loss and metric calculated differently on validation

### ISSUE THREAD

**Issue by michalCyberfish** [tensorflow/tensorflow#43763]
I'm using `keras.applications.vgg19` with some custom loss function (cross-entropy based). I'm using the loss function both as a loss and as a metric. While training, I've been getting the same values for both loss and metric, but for validation, those values are different.

![image](https://user-images.githubusercontent.com/59140738/95012520-ecc66d00-0641-11eb-99e0-370ff41af92f.png)

You can see that for training, the loss and loss metric are the same (orange and blue lines that merge into one line), and for validation the loss and loss metric look different.

```
Epoch 1/7
 - 111s - loss: 3.3320 - loss_metric: 3.3320 - val_loss: 3.0186 - val_loss_metric: 3.0993 
Epoch 2/7
 - 90s - loss: 3.0934 - loss_metric: 3.0934 - val_loss: 3.0394  - val_loss_metric: 3.0424 
Epoch 3/7
 - 91s - loss: 3.0643 - loss_metric: 3.0643 - val_loss: 3.0139  - val_loss_metric: 3.0031 
Epoch 4/7
 - 90s - loss: 3.0411 - loss_metric: 3.0411 - val_loss: 3.0588  - val_loss_metric: 2.9770 
Epoch 5/7
 - 90s - loss: 3.0170 - loss_metric: 3.0170 - val_loss: 2.8466  - val_loss_metric: 2.9625 
Epoch 6/7
 - 90s - loss: 2.9866 - loss_metric: 2.9866 - val_loss: 3.0236  - val_loss_metric: 2.9133 
Epoch 7/7
 - 91s - loss: 2.9447 - loss_metric: 2.9447 - val_loss: 2.9063  - val_loss_metric: 3.0723
```

Some notes:

- I know this might be due to the usage of regularization, however, I'm not finding any regularization in the  [Keras implementation of VGG19](https://github.com/keras-team/keras-applications/blob/master/keras_applications/vgg19.py)
- I know there a was a [bug ](https://github.com/tensorflow/tensorflow/issues/25970#issuecomment-470210090)which has been fixed for TF2. I'm using TensorFlow GPU 2.1.0 and Keras 2.3.1

Can anyone please tell me what am I missing?

**Comment by amahendrakar** [comment]
@michalCyberfish,
This question is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow) since it is not a TensorFlow bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by jvishnuvardhan** [comment]
@michalCyberfish Can you please share a simple standalone code to reproduce the issue? Thanks!

**Comment by michalCyberfish** [comment]
>
> train_datagen = ImageDataGenerator(rescale=1./255,)
> train_generator= train_datagen .flow_from_dataframe(
>
directory="./train/",

x_col="id",

y_col="label",

batch_size=32,

shuffle=True,

class_mode="categorical",

target_size=(32,32))

dataframe=train_df,


this prints this line for each flow_from_dataframe initialization:

> Found 2000 images belonging to 2 classes


which is really annoying when you use it inside another generator. and it
prints it thousand of times.

**Comment by jvishnuvardhan** [comment]
@michalCyberfish Generally you need to use `ImageDataGenerator` once to load generator. So  ` Found 2000 images belonging to 2 classes` this should be printed only one time. Please check [this tutorial](https://colab.research.google.com/github/tensorflow/docs/blob/master/site/en/tutorials/images/transfer_learning_with_hub.ipynb#scrollTo=0p7iDOhIcqY2) where `ImageDataGenerator` was used.

Anyway, how is this print out related to the title of this issue? Can you please update it so that other users will find it useful. 

If there are any further questions, please provide a simple standalone code to reproduce the issue. Thanks!

**Comment by michalCyberfish** [comment]
I’m sorry, I thought you replied on the other issue I opened. I will
provide a standalone code for this issue shortly.

**Comment by jvishnuvardhan** [comment]
@michalCyberfish I am closing this issue. I think the tutorial i mentioned in the last comment should help understanding usage of `ImageDataGenerator`.

If you still face any issue, Please feel free to reopen with a simple standalone code to reproduce the issue. Thanks!
